docs/faq.md

```diff
@@ -0,0 +1,89 @@
+# Frequently Asked Questions
+
+## What is Jupytext?
+
+Jupytext is a Python package that can convert Jupyter notebooks to scripts or Markdown documents. It can also convert these text documents back to Jupyter notebooks.
+
+## Why would I want to convert my notebooks to text?
+
+The text representation have much cleaner diffs than the original notebook format. Merging multiple contributions to a notebook in any of these text formats is easier than with the JSON format. Last but not least, acting on a notebook represented as text (spell check, reformat, ...) is sometimes more comfortable than in Jupyter.
+
+## How do I use Jupytext?
+
+Open the notebook you want to version control. _Pair_ the notebook to a script or a Markdown file using either the [Jupytext Menu](https://github.com/mwouts/jupytext/blob/master/README.md#jupytext-menu-in-jupyter-notebook) in Jupyter Notebook or the [Jupytext Commands](https://github.com/mwouts/jupytext/blob/master/README.md#jupytext-commands-in-jupyterlab) in JupyterLab.
+
+Save the notebook, and you get two copies of the notebook: the original `*.ipynb` file, together with its paired text representation.
+
+## Which Jupytext format do you recommend?
+
+I tend to use the Markdown format for notebooks that contain more text than code, as Markdown documents are conveniently edited in IDEs and also well rendered on GitHub.
+
+Saving notebooks as scripts is a convenient choice when you want to refactor your notebook in an IDE (or import it in another notebook, etc). Use the `percent` format if you prefer to get explicit cell markers (compatible with VScode, PyCharm, Spyder, Hydrogen...). If you prefer to get the minimal amount of cell markers, go for the `light` format.
+
+## Can I see a sample of each format?
+
+Go to [our demo folder](https://github.com/mwouts/jupytext/tree/master/demo) and see how our sample `World population` notebook is represented in each format.
+
+## Can I edit the paired text file?
+
+Yes! And when you're done, refresh the notebook in Jupyter. Refreshing will bring the latest changes to your notebook.
```

Comment:
Will this work even if the text file is edited when a jupyter server isn't running?

e.g., if I have a notebook and paired markdown file, I synchronize them, and push them both to GitHub. Then somebody else updates *just the markdown file* and pushes the changes to GitHub. I pull in the changes, and turn on JupyterLab...what happens?

Reply:
> Will this work even if the text file is edited when a jupyter server isn't running?

Indeed! You're free to close the server. There's no magic here: when the notebook is opened or refreshed, Jupytext reads the two files and merges inputs+outputs. The ipynb file is not modified when the notebook is read, but only the next time it is saved.

If you keep the notebook open, the extra bonus is that variables are preserved when you refresh the notebook. I'll add a question about that.

> e.g., if I have a notebook and paired markdown file, I synchronize them, and push them both to GitHub. Then somebody else updates just the markdown file and pushes the changes to GitHub. I pull in the changes, and turn on JupyterLab...what happens?

What happens is what you expect: you get the latest input cells from the markdown file, matched with outputs from the ipynb file. It will work 100% if you don't push the ipynb file to GitHub, and only 99% (*) if you do push the ipynb file.

Let me explain: Jupytext in Jupyter is very strict about the assumption that Jupyter always writes the ipynb file *before* the md file. If git happens to write the ipynb more than [one second](https://github.com/mwouts/jupytext/blob/872c6585738d4df252c41b21feb40a19b95a1c8b/jupytext/contentsmanager.py#L131-L136) after the md file, Jupytext will complain and refuse to open the notebook.